tree/ntuple/v7/src/RPageStorageRoot.cxx

```diff
@@ -240,11 +240,11 @@ ROOT::Experimental::Detail::RPageSourceRoot::RPageSourceRoot(std::string_view nt
 {
 }
 
-ROOT::Experimental::Detail::RPageSourceRoot::RPageSourceRoot(std::string_view ntupleName, std::string_view path)
+ROOT::Experimental::Detail::RPageSourceRoot::RPageSourceRoot(std::string_view ntupleName, std::string_view path, TDirectory* directory)
    : RPageSource(ntupleName)
    , fPageAllocator(std::make_unique<RPageAllocatorKey>())
    , fPagePool(std::make_shared<RPagePool>())
-   , fDirectory(nullptr)
+   , fDirectory(directory)
```

Comment:
It still works perfectly although it might seem strange. Because the newly overwritten directory and the old one both originate from the same .root file.

Reply:
One possibility to solve it would be:
`if (!fDirectory) {
   fDirectory = fSettings.fFile->GetDirectory(fNTupleName.c_str());
}`
but I would leave it because it may affect other parts of code.